apps/judicial-system/backend/src/app/modules/case/case.controller.ts

```diff
@@ -179,7 +181,22 @@ export class CaseController {
       )
     }
 
-    return this.caseService.update(theCase, caseToUpdate, user) as Promise<Case> // Never returns undefined
+    const update = this.caseService.update(
+      theCase,
+      caseToUpdate,
+      user,
+    ) as Promise<Case>
+
+    return Promise.all([
+      update,
+      ...(caseToUpdate.courtCaseNumber && theCase.state === CaseState.SUBMITTED
+        ? [
+            this.transition(caseId, user, theCase, {
+              transition: CaseTransition.RECEIVE,
+            }),
+          ]
+        : []),
+    ]).then((values) => values[0])
```

Comment:
This implementation will not perform both updates in a single transaction. Although less likely to cause problems, this does not completely fix the issue we are trying to prevent.

Consider modifying the code to perform the two updates in a single database transaction in the case service.